### DISCUSSION TITLE
How to ensure all db operations in a same transaction?

### DISCUSSION THREAD

**Discussion by formingform** [blockscout/blockscout#9458] [Q&A] [answered]
I tried to implement a fetcher to filter my contract events, and I'll import the events to db like this:
```
Chain.import(events_data)
```
But, before to import events to db, I'll do something else and update my business tables, so, how can I ensure "import events" and "update business tables" in a same transaction?

Thanks.

**Comment by sl1depengwyn** [comment, marked as answer]
In blockscout we do it like this https://hexdocs.pm/ecto/Ecto.Multi.html

**Reply by formingform** [reply]
thanks
